src/feelpp/benchmarking/reframe/report.py

```diff
@@ -41,50 +33,91 @@ class Report:
     def __init__(self, file_path):
 
         self.file_path = file_path
-
         self.partial = False
         if os.path.basename(self.file_path) not in notPartial:
             self.partial = True
 
-        # Following attributes will be set using methods
+        # Following attributes will be set while processing the data
         self.data               = None
+        self.toolbox            = ''
         self.ref_speedup        = -1
         self.df_perf            = pd.DataFrame()
         self.df_speedup         = pd.DataFrame()
         self.df_partialPerf     = pd.DataFrame() if self.partial else None
         self.df_partialSpeedup  = pd.DataFrame() if self.partial else None
+        self.partialDict        = dict() if self.partial else None
 
         self.load()
         self.processData()
 
 
-
     def load(self):
         with open(self.file_path, 'r') as file:
             self.data = json.load(file)
 
 
     def processData(self):
+        # Avoids error for reporting scale performances not launched with 'CpuVariation.py'
+        try:
+            self.toolbox = self.data['runs'][0]['testcases'][0]['check_vars']['toolbox']
+        except KeyError as e:
+            self.toolbox = ''
+
         # Needed dataframe for building df_perf and df_speedup
         df = pd.DataFrame(self.data['runs'][0]['testcases'][0:])
         df['num_tasks'] = df['check_vars'].apply(lambda x: x['num_tasks'])
         df['num_tasks'] = df['num_tasks'].astype(int)
         df = df[['num_tasks', 'perfvars']]
 
+        if self.partial:
+            self.buildPartialDict()
         self.buildPerf(df)
         self.buildSpeedup(df)
+        self.linearReg(self.df_speedup)
+        if self.partial:
+            self.linearReg(self.df_partialSpeedup)
+
+
+    def buildPartialDict(self):
+        tmpDict = {}
+        for perfvar in self.data['runs'][0]['testcases'][0]['perfvars']:
+            name = perfvar['name']
+            stageName = name.split('_')[0]
 
+            if stageName not in tmpDict.keys():
+                tmpDict[stageName] = []
+            tmpDict[stageName].append(name)
+
+        self.partialDict = {}
+        for stageName, perfNameLst in tmpDict.items():
+            perfNameLst.pop()
+            perfNameLst = [name.split('_')[1] for name in perfNameLst]
+            self.partialDict.update( {stageName : perfNameLst} )
+
+
+    def isPartialPerf(self, perfName):
+        if self.partial:
+            for nameLst in self.partialDict.values():
+                for name in nameLst:
+                    if name in perfName:
+                        return True
+            return False
+        else:
+            return False
 
 
     def buildPerf(self, df):
         for k, t in enumerate(df['num_tasks'].unique()):
             for perf in df['perfvars'][k]:
```

Comment:
Fix the case where `df["perfvars"]==None` . Maybe skip the function and just keep to total execution time per testcase in order to plot a speedup (It is the case for old KUB reframe reports, we sill want to have the general speedup of the app).

Reply:
to do while refracting in [#72](https://github.com/feelpp/benchmarking/issues/72)